calicoctl/commands/resources.go

```diff
@@ -124,6 +124,16 @@ func getResourceFromArguments(args map[string]interface{}) (unversioned.Resource
 			p.Metadata.CIDR = *cidr
 		}
 		return *p, nil
+	case "bgpPeer":
+		p := api.NewBGPPeer()
+		if name != "" {
+			err := p.Metadata.PeerIP.UnmarshalText([]byte(name))
+			if err != nil {
+				return nil, err
+			}
+		}
+		p.Metadata.Hostname = hostname
```

Comment:
BGPPeer does not have a Name field.  It's primary identifier is the IP address.

Now, the get command syntax currently assumes that every object is identified by a name field ... which it isn't.  :-(   So I think some work is actually required to reconsider what the shape of a generic get command should be.  I'll raise this as an issue to make sure I don't forget about it - but I *think* this will be a post-alpha thing.  Maybe we could have a quick brain storm about that.